specs/PushNotifications/ToastNotifications-spec.md

```diff
@@ -0,0 +1,551 @@
+winrt::Windows::Foundation::IAsyncAction UpdateProgressAsync()
+{
+    winrt::hstring tag = L"weekly-playlist";
+    winrt::hstring group = L"downloads";
+
+    // Assign new values
+    // Note that you only need to assign values that changed. In this example we don't assign progressStatus since we don't need to change it
+    AppNotificationProgressData data;
+    data.Value(0.7); // Binds to {progressValue} in xml payload
+    data.ValueStringOverride(L"18/26 songs"); // Binds to {progressValueString} in xml payload
+    data.SequenceNumber(2);
+
+    auto result = co_await AppNotificationManager::Default().UpdateProgressDataAsync(data, tag, group);
+    if (result == AppNotificationProgressResult::AppNotificationNotFound)
+    {
+        LOG_HR_MSG(E_NOT_SET, "Toast Progress Update Failed since the previous notification update is missing or has been cleared by the user!");
+    }
+}
+```
+# Remarks
+
+## Registration
+
+The developer should always call the Notification Registration API first to register the current process as
+the Activator target.
+
+## Foreground API calls
+
+The developer should always subscribe to AppNotification activation foreground events to intercept notification related actions if
+the app happens to be running in the foreground.
+
+## Manifest Registration
+
+For MSIX, the COM activator GUID and the exe need to be registered in the manifest. The launch args
+would need to be pre-set to a well-known string that defines Activation Triggers.
+
+```xml
+<Extensions>
+  <com:Extension Category="windows.comServer">
+    <com:ComServer>
+      <com:ExeServer Executable="SampleBackgroundApp\SampleBackgroundApp.exe" DisplayName="SampleBackgroundApp" Arguments="----AppNotificationActivationServer">
+        <com:Class Id="{GUIDEntryPoint}" />
+      </com:ExeServer>
+    </com:ComServer>
+  </com:Extension>
+</Extensions>
+```
+
+# API Details
+
+```c#
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+import "..\AppLifecycle\AppLifecycle.idl";
+
+namespace Microsoft.Windows.AppNotifications
+{
+    // The Registration Info for Packaged and Unpackaged Notification Activations
+    runtimeclass AppNotificationActivationInfo
+    {
+        // Initialize using a manifest defined COM Activator Id. Only applicable to Packaged Win32 applications
+        AppNotificationActivationInfo(Guid taskClsid);
+
+        // Initialize using Notification Assets. Only applicable to Unpackaged Win32 applications which need to specify their own assets like DisplayName and Icon.
+        AppNotificationActivationInfo(String displayName, Windows.Foundation.Uri iconUri);
+
+        // The CLSID associated with the Client COM server that Windows App SDK will activate
+        Guid TaskClsid{ get; };
+
+        // The App friendly DisplayName for the Notification in Action Centre 
+        String DisplayName{ get; };
```

Comment:
>DisplayName [](http://example.com/codeflow?start=15&length=11)

How is the unpakaged app identified when it launches a second time, so we can tell it about the notifications from previous instances that are still visible in Action Center? Do the DisplayName+IconUri both have to match? Only the DisplayName? Only the IconUri? Executable path? Do they never match and we assume every instance is a new app unrelated to any previous instance?

Reply:
We use the Process's full file path. By design: We will lose the notifications if the app decides to change the path in any way.